<launch>
    <arg name="map_size_x_"/>
    <arg name="map_size_y_"/>
    <arg name="map_size_z_"/>

    <arg name="cx"/>
    <arg name="cy"/>
    <arg name="fx"/>
    <arg name="fy"/>

        <!-- maximum velocity and acceleration the drone will reach -->
    <arg name="max_vel"/>
    <arg name="max_acc"/>
    <arg name="max_jerk"/>

    <arg name="planning_horizon"/>

    <!-- main node -->
    <node pkg="ego_planner" name="ego_planner_node" type="ego_planner_node" output="screen" launch-prefix="gnome-terminal --">
<!--        <node pkg="ego_planner" name="ego_planner_node" type="ego_planner_node" output="screen">-->
        <!-- planning fsm -->
        <!-- time to replan, the less, the higher frequency -->
        <param name="fsm/thresh_replan" value="0.05" type="double"/>
        <!-- distance to stop replan -->
        <param name="fsm/thresh_no_replan" value="0.2" type="double"/>
        <!--always set to 1.5 times grater than sensing horizen-->
        <param name="fsm/planning_horizon" value="$(arg planning_horizon)" type="double"/>
        <param name="fsm/emergency_time_" value="1.0" type="double"/>

        <param name="grid_map/resolution"      value="0.1" />
        <param name="grid_map/map_size_x"   value="$(arg map_size_x_)" />
        <param name="grid_map/map_size_y"   value="$(arg map_size_y_)" />
        <param name="grid_map/map_size_z"   value="$(arg map_size_z_)" />
        <param name="grid_map/local_update_range_x"  value="4.0" />
        <param name="grid_map/local_update_range_y"  value="4.0" />
        <param name="grid_map/local_update_range_z"  value="2.0" />
        <param name="grid_map/obstacles_inflation"     value="0.35" />
        <param name="grid_map/local_map_margin" value="10"/>
        <param name="grid_map/ground_height"        value="-0.01"/>

        <!-- camera parameter -->
            <!-- intrinsic params of the depth camera -->
        <param name="grid_map/cx" value="$(arg cx)"/>
        <param name="grid_map/cy" value="$(arg cy)"/>
        <param name="grid_map/fx" value="$(arg fx)"/>
        <param name="grid_map/fy" value="$(arg fy)"/>
        <!-- depth filter -->
        <param name="grid_map/use_depth_filter" value="true"/>
        <param name="grid_map/depth_filter_tolerance" value="0.15"/>
        <param name="grid_map/depth_filter_maxdist"   value="5.0"/>
        <param name="grid_map/depth_filter_mindist"   value="0.2"/>
        <param name="grid_map/depth_filter_margin"    value="1"/>
        <param name="grid_map/k_depth_scaling_factor" value="1000.0"/>
        <param name="grid_map/skip_pixel" value="2"/>
        <!-- local fusion -->
        <param name="grid_map/p_hit"  value="0.65"/>
        <param name="grid_map/p_miss" value="0.35"/>
        <param name="grid_map/p_min"  value="0.12"/>
        <param name="grid_map/p_max"  value="0.90"/>
        <param name="grid_map/p_occ"  value="0.80"/>
        <param name="grid_map/min_ray_length" value="0.1"/>
        <param name="grid_map/max_ray_length" value="4.5"/>

        <param name="grid_map/virtual_ceil_height"   value="2.99"/>
        <param name="grid_map/visualization_truncate_height"   value="2.9"/>
        <param name="grid_map/show_occ_time"  value="false"/>
        <!-- 1: pose; 2: odom -->
        <param name="grid_map/pose_type"     value="2"/>
        <param name="grid_map/frame_id"      value="map"/>

        <!-- planner manager -->
        <param name="manager/max_vel" value="$(arg max_vel)" type="double"/>
        <param name="manager/max_acc" value="$(arg max_acc)" type="double"/>
        <param name="manager/max_jerk" value="$(arg max_jerk)" type="double"/>
        <param name="manager/control_points_distance" value="0.4" type="double"/>
        <param name="manager/feasibility_tolerance" value="0.05" type="double"/>
        <param name="manager/planning_horizon" value="$(arg planning_horizon)" type="double"/>

        <!-- trajectory optimization -->
        <param name="optimization/lambda_smooth" value="1.0" type="double"/>
        <param name="optimization/lambda_collision" value="0.5" type="double"/>
        <param name="optimization/lambda_feasibility" value="0.1" type="double"/>
        <param name="optimization/lambda_fitness" value="1.0" type="double"/>
        <param name="optimization/dist0" value="0.5" type="double"/>
        <param name="optimization/max_vel" value="$(arg max_vel)" type="double"/>
        <param name="optimization/max_acc" value="$(arg max_acc)" type="double"/>
<!--        <param name="optimization/max_jerk" value="$(arg max_jerk)" type="double"/>-->

        <param name="bspline/limit_vel" value="$(arg max_vel)" type="double"/>
        <param name="bspline/limit_acc" value="$(arg max_acc)" type="double"/>
        <param name="bspline/limit_ratio" value="1.1" type="double"/>
    </node>
</launch>
